src/onehot.jl

```diff
@@ -83,20 +83,36 @@ julia> reshape(1:15, 3, 5) * oh  # this matrix multiplication is done efficientl
 """
 onehotbatch(data, labels, default...) = _onehotbatch(data, length(labels) < 32 ? Tuple(labels) : labels, default...)
 
-function _onehotbatch(data, labels)
-  indices = UInt32[something(_findval(i, labels), 0) for i in data]
-  if 0 in indices
+function _onehotbatch(data, labels)  # this accepts any iterator
+  indices = UInt32[something(_findval(x, labels), 0) for x in data]
+  if any(iszero, indices)
     for x in data
-      isnothing(_findval(x, labels)) && error("Value $x not found in labels")
+      isnothing(_findval(x, labels)) && throw(ArgumentError("Value x = $x not found in labels = $labels"))
     end
   end
   return OneHotArray(indices, length(labels))
 end
+function _onehotbatch(data::AbstractArray, labels)  # this works for GPUArrays too
```

Comment:
One reason to change this is to avoid ever making a `MVector` or something weird like that:
```suggestion
function _onehotbatch(data::AbstractGPUArray, labels)
```